include/seqan3/std/ranges

```diff
@@ -13,8 +13,45 @@
 
 #pragma once
 
-#if __cpp_lib_ranges // C++20 ranges available
+#if __has_include(<ranges>)
 #include <ranges>
+#endif // __has_include(<ranges>)
+
+#if defined(__cpp_lib_ranges) // C++20 ranges available
```

Comment:
Does `__has_include(<ranges>)` imply `defined(__cpp_lib_ranges)`? 

If yes, why did you move the `<ranges>` include outside of this if condition? (This looks like the `<ranges>` include is independent of all the following code but before the else belonged to the `<ranges>` include if clause.)
If no, there is a possibility that we include the ranges header **and** the still include the `range/v3/...` headers (else clause), if I understand it right? Couldn't this potentially cause errors?

Reply:
In C++17 mode the header exists, but does not contain anything.

We could also include `<version>` and would get the macro `__cpp_lib_ranges`.

Since we use the same construct in `std/span` I kept the old approach to first include the header (which might be empty) and after that use the feature macro to make sure it exists.

Reply:
> In C++17 mode the header exists, but does not contain anything.

why? o.O

I still think it is confusing that way, I also don't like it in span. Can it not happen that <ranges> has some content but not everything to have `__cpp_lib_ranges` set? 
If not, I guess just leave it.

Reply:
> I still think it is confusing that way, I also don't like it in span. Can it not happen that has some content but not everything to have `__cpp_lib_ranges` set?

This is something we can never make sure. If someone only half-implements something and falsely advertises via `__cpp_lib_ranges`, which is the same as saying "Hi I'm able to support ranges completely", we are always screwed.



> > In C++17 mode the header exists, but does not contain anything.
> 
> why? o.O

Think about it: The includes are always stored in one location. The only difference is that gcc will enable / disable certain behaviours in the header depending on the C++ mode. If you are requesting C++17 you don't want to pull in any concepts stuff.

That means a existing header is not the same as "it is also available and usable".